documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -261,4 +263,302 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
+
+### <a name="R2013" />R2013 XmsClientNameProperty
+**Output Message**: Value of 'x-ms-client-name' cannot be the same as '{0}' Property/Model.
+
+**Description**: The [`x-ms-client-name`](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-client-name) extension is used to change the name of a parameter or property in the generated code. By using the 'x-ms-client-name' extension, a name can be defined for use specifically in code generation, separately from the name on the wire. It can be used for query parameters and header parameters, as well as properties of schemas.
+
+**Why the rule is important**: This value cannot be same as parameter name or property name, because having the same name invalidates the usage.
+
+**How to fix the violation**: Please use different name that you'd like to see on the generated code.
+
+**Impact on generated code**: Generated SDK code will expose the corresponding client name on property or model.
+
+**Examples**:
+```
+{
+  "definitions": {
+    "Product": {
+      "x-ms-external" : true,
+      "properties": {
+        "product_id": {
+          "type": "string",
+          "x-ms-client-name": "SKU"
+        }
+     }
+  }
+}
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R1010" />R1010 AvoidMsdnReferences
+**Output Message**: For better generated code quality, remove all references to "msdn.microsoft.com".
+
+**Description**: The documentation is being generated from the swagger and published at "docs.microsoft.com". From that perspective, documentation team would like to avoid having links to the "msdn.microsoft.com" in the swagger and SDK documentations.
+
+**Why the rule is important**: Facilitate decoupling of "msdn.microsoft.com" from "docs.microsoft.com".
+
+**How to fix the violation**: Please avoid using referenes to "msdn.microsoft.com" in title and descriptions.
+
+**Impact on generated code**: N/A.
+
+**Examples**: N/A.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R1009" />R1009 DeleteInOperationName
+**Output Message**: 'DELETE' operation '{0}' should use method name 'Delete'. Note: If you have already shipped an SDK on top of this spec, fixing this warning may introduce a breaking change.
+
+**Description**: Verifies whether value for `operationId` is named as per ARM guidelines.
+
+**Why the rule is important**: Per ARM guidelines, each 'DELETE' operation on a resource should have "delete" in the name.
```

Comment:
I'm not sure this is per ARM guidelines, maybe more like ARM SDK guidelines? Guidelines are in place for a more consistent customer experience with ARM  SDKs, so may be want to mention that here, for consistency among ARM services SDKs.